ufm-test-services/metamask/tests/prometheusUtils.ts

```diff
@@ -0,0 +1,70 @@
+import 'dotenv/config'
+import fetch from 'node-fetch'
+import { Counter, Pushgateway } from 'prom-client'
+
+const txSuccessMetricName = 'metamask_tx_success'
+const txFailureMetricName = 'metamask_tx_failure'
+
+const txSuccessCounter = new Counter({
+  name: txSuccessMetricName,
+  help: 'A counter signifying the number of successful transactions sent with Metamask since last failure',
+})
+const txFailureCounter = new Counter({
+  name: txFailureMetricName,
+  help: 'A counter signifying the number of failed transactions sent with Metamask since last successful transaction',
+})
+
+export const getMetricValueFromPrometheus = async (metricName: string) => {
+  if (process.env.PROMETHEUS_SERVER_URL === undefined)
+    throw new Error('Please provied PROMETHEUS_SERVER_URL in .env')
+
+  const response = await fetch(
+    `${process.env.PROMETHEUS_SERVER_URL}/api/v1/query?query=${metricName}`
+  )
+  const reponseJson = (await response.json()) as {
+    status: string
+    data: { resultType: string; result: any[] }
+  }
+
+  if (
+    reponseJson &&
+    reponseJson.data &&
+    reponseJson.data.result &&
+    reponseJson.data.result.length > 0
+  ) {
+    return parseFloat(reponseJson.data.result[0].value[1])
+  } else {
+    console.warn(`No data found for metric ${metricName} in Prometheus`)
+    return undefined
+  }
+}
```

Comment:
If you use zod this would already be validated. You can also use .transform to parse the float